device-to-group-command.py

```diff
@@ -0,0 +1,404 @@
+
+
+def whitelist_package_in_group_devices(device_list, package_name):
+    # Fetch all devices in group
+
+    # None check
+    print("Currently active devices in group: ")
+    print([device.device_name for device in device_list])
+
+    for device in device_list:
+        if not is_package_whitelisted(device.id, package_name):
+            try:
+                run_whitelist_command(device, package_name)
+            except Exception as e:
+                print('Could not fire command on ' + device.device_name + ": " + str(e))
+            time.sleep(1)
+        else:
+            print('Warning: Package already whitelisted or not present on device: ' + device.device_name)
+
+
+# ====== UNINSTALL FROM GROUP ==========            
+
+
+def run_uninstall_command(device, package_name, app_id):
+    api_instance = esperclient.CommandsApi(esperclient.ApiClient(configuration))
+    command_args = esperclient.CommandArgs(package_name=package_name, app_version = app_id)
+    command = esperclient.CommandRequest(command='UNINSTALL', command_args=command_args)
+    #print(command)
+    try:
+        api_response = api_instance.run_command(enterprise_id, device.id, command)
+        if api_response.id is not None:
+            print('Uninstall command fired successfully on ' +  device.device_name + ' for ' + package_name)
+    except ApiException as e:
+        print("Exception when calling CommandsApi->run_command: %s\n" % e)
+
+
+def uninstall_package_in_group_devices(device_list, package_name):
+    # Fetch all devices in group
+
+    print("Currently active devices in group: ")
+    print([device.device_name for device in device_list])
+
+    for device in device_list:
+        app_id = is_package_installed(device.id, package_name)
+        if app_id:
+            try:
+                run_uninstall_command(device, package_name, app_id)
+            except Exception as e:
+                print('Could not fire command on ' + device.device_name + ": " + str(e))
+            time.sleep(1)
+        else:
+            print('Warning: Package already uninstalled or not present on device: ' + device.device_name)
+
+def is_package_installed(device_id, package_name):
+    querystring = {"package_name": package_name, "whitelisted":"false"}
+    payload = ""
+    headers = {
+        'Authorization': "Bearer " + configuration.api_key['Authorization'],
+        'cache-control': "no-cache",
+        }
+
+    api_instance = esperclient.DeviceApi(esperclient.ApiClient(configuration))
+    response = api_instance.get_app_installs(enterprise_id, device_id, package_name=package_name)
+    #print(response)
+    if response and response.count > 0:
+        app_id = response.results[0].application.version.app_version_id
+        if app_id:
+          return response.results[0].application.version.app_version_id
+        else:
+            return 0
+    else:
+        #print("App doesn't exist")
+        return 0
```

Comment:
(nit) you can just say return 0, no else required. Also, function says `is_package_whitelisted` which expects it to return true or false value. However you are returning app_id if present. Better to change function name then.

Reply:
Done